### PR TITLE
Next graph-level of ROS packages

### PR DESCRIPTION
**Pull request by wolfv**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by wolfv** [comment]
@seanyen, again, I made you a maintainer here

**Comment by seanyen** [comment]
> @seanyen, again, I made you a maintainer here

Sounds good to me. :)

File: recipes/class_loader/build.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# turn this into a catkin workspace by adding the catkin token if it doesn't
+# exist already
+if [ ! -f $PREFIX/.catkin ]; then
+    touch $PREFIX/.catkin
+fi
+
+# necessary for correctly linking SIP files (from python_qt_bindings)
+export LINK=$CXX
+
+export ROS_PYTHON_VERSION=`${PYTHON} -c "import sys; print('%i.%i' % (sys.version_info[0:2]))"`
```
Comment (xhochy) [inline recipes/class_loader/build.sh]: Is this only needed during the build or would this also be useful as an activation script?

File: recipes/gencpp/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    - python
+    - ros-conda-mutex * melodic
+    - ros-conda-base
+    - ros-genmsg
+
+test:
+  requires:
+    - rospkg
+  commands:
+    - export ROS_ROOT=$PREFIX && python -c "import rospkg;s = rospkg.RosPack();ret = s.get_path('gencpp');print(ret)"
```
Comment (xhochy) [inline recipes/gencpp/meta.yaml]: Could `ROS_ROOT` also go into an activation script?

File: recipes/class_loader/build.sh (hunk 1 above)
Comment (wolfv) [reply recipes/class_loader/build.sh]: I think it's safer to set it like this. It might be set in the ros_environment package.

**Comment by wolfv** [comment]
> Looks good but also seems like a lot of duplication. Would it be worth to put the build scripts in a meta-package? Also there seems to be general environment variables. Instead of setting them manually could they be part of an activation script in e.g. `ros-conda-base` or `ros-environment`?

As mentioned, all environment vars that are really required at runtime should be set in ros-catkin and ros-environment.

I think the ones that are set in the build script are fine there as they are somewhat obscure (especially the $LINK one).

In any case these recipes are automatically generated, and I tried to come up with the universal build process that will work for all packages.

I will think about the idea with the ros-conda-build package that contains the build script, that could indeed be a nice simplification!

**Pull request merged**